src/openrct2/management/research.c

```diff
@@ -208,12 +211,40 @@ void research_finish_item(sint32 entryIndex)
 				}
 			}
 
-			set_format_arg(0, rct_string_id, ((rideEntry->flags & RIDE_ENTRY_FLAG_SEPARATE_RIDE_NAME)) ?
-				rideEntry->name : RideNaming[base_ride_type].name);
+			if (!gConfigInterface.select_by_track_type) {
+				availabilityString = STR_NEWS_ITEM_RESEARCH_NEW_RIDE_AVAILABLE;
 
+				set_format_arg(0, rct_string_id, ((rideEntry->flags & RIDE_ENTRY_FLAG_SEPARATE_RIDE_NAME)) ?
+							   rideEntry->name : get_friendly_track_type_name(base_ride_type, rideEntry));	// Makes sure the correct track name is displayed,
+																											// e.g. Hyper-Twister instead of Steel Twister.
+			} else {
+				// If a vehicle is the first to be invented or does not want to be separate, show the ride group name.
```

Comment:
Needs checking. What if Cheshire Cats come before Car Ride? Having a way to determine if a ride group was invented is probably better.